cachecontrol/controller.py

```diff
@@ -117,6 +117,12 @@ def cached_request(self, request):
             return resp
 
         headers = CaseInsensitiveDict(resp.headers)
+        if not headers or 'date' not in headers:
+            # With date or etag, the cached response can never be used
+            # and should be deleted.
+            if 'etag' not in headers:
```

Comment:
I don't think that's correct.

This method has to return a cached response or ``False``. With the returned value of this, the cachecontrol adapter will either use the returned cached response without hitting the network, or it'll hit the network. Since there's no date header we can never return an object that doesn't have a date header associated with it since we can't determine what the fresh-ness is.

When this returns False the adapter will do some network, if the ETag is fine it'll do the update_cached_response call and load the cached item then outside of this method.